routes/tp.js

```diff
@@ -328,15 +310,79 @@ router.put('/tp/:id/reserve/now', async function(req, res) {
         newresa = await tpssrv.create_tp_reservation(reservation_id, 'auto');
     } catch (error) {
         logger.error(error);
-        res.status(500).send({message: 'Error will creating tp reservation'});
-        res.end();
+        res.status(500).send({message: 'Error while creating tp reservation'});
         return;
     }
 
     logger.debug('set reservation as done', newresa);
     newresa.created = true;
     res.send({reservation: newresa});
-    res.end();
+});
+
+router.put('/tp/:id/reserve/extend', async function(req, res) {
+    if(! req.locals.logInfo.is_logged) {
+        res.status(403).send({message: 'Not authorized'});
+        return;
+    }
+    if(! sansrv.sanitizeAll([req.params.id])) {
+        res.status(403).send({message: 'Invalid parameters'});
+        return;
+    }
+    let user = null;
+    let isadmin = false;
+    try {
+        user = await dbsrv.mongo_users().findOne({'_id': req.locals.logInfo.id});
+        isadmin = await rolsrv.is_admin(user);
+    } catch(e) {
+        logger.error(e);
+        res.status(404).send({message: 'User session not found'});
+        return;
+    }
+
+    if(!user) {
+        res.status(404).send({message: 'User does not exist'});
+        return;
+    }
+
+    if(!isadmin) {
+        res.status(403).send({message: 'Not authorized'});
+        return;
+    }
+
+    let reservation_id = ObjectID.createFromHexString(req.params.id);
+
+    let filter = {};
+    if(isadmin) {
+        filter = {_id: reservation_id};
+    }
+    else{
+        filter = {_id: reservation_id, owner: user.uid};
+    }
+    let reservation = await dbsrv.mongo_reservations().findOne(filter);
+    if(!reservation){
+        res.status(403).send({message: 'Not allowed to extend this reservation'});
```

Comment:
Since only admins can extend, this can only mean the reservation does not exists (with this ID). Should clarify the error message